src/cdk/schematics/ng-update/devkit-file-system.ts

```diff
@@ -73,4 +62,23 @@ export class DevkitFileSystem extends FileSystem<Path> {
     const buffer = this._tree.read(filePath);
     return buffer !== null ? buffer.toString() : null;
   }
+
+  existsDirectory(dirPath: Path) {
```

Comment:
Couldn't we bake this into `exists` which would handle both directories and files? I think that's what Node does.

Reply:
I considered it, but wanted to avoid it as that matches with TypeScript's API. I'll play with it. Generally I'd agree that this would be nicer.